### PR TITLE
atmos 1.3.9 (new formula)

### PR DESCRIPTION
**Pull request by nitrocode**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

```
brew install --build-from-source atmos
brew test atmos
brew audit --strict atmos
```

## References
* https://github.com/cloudposse/atmos

### CODE REVIEW

File: Formula/atmos.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  sha256 "dabae8c89e508bd150640ec562f2642854c50a8e6ffbfd881a701ae40394f533"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOARCH"] = Hardware::CPU.intel? ? "amd64" : Hardware::CPU.arch.to_s
+    ENV["GOOS"] = OS.mac? ? "darwin" : "linux"
+    # system "go", "get"
+    system "go build -o build/atmos -v -ldflags \"-X 'github.com/cloudposse/atmos/cmd.Version=test3'\""
```
Comment (SMillerDev) [inline Formula/atmos.rb:14]:
```suggestion
    system "go build *std_go_args(ldflags: "-X 'github.com/cloudposse/atmos/cmd.Version=#{version}'")
```

File: Formula/atmos.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  homepage "https://github.com/cloudposse/atmos"
+  url "https://github.com/cloudposse/atmos/archive/refs/tags/v1.3.9.tar.gz"
+  sha256 "dabae8c89e508bd150640ec562f2642854c50a8e6ffbfd881a701ae40394f533"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOARCH"] = Hardware::CPU.intel? ? "amd64" : Hardware::CPU.arch.to_s
+    ENV["GOOS"] = OS.mac? ? "darwin" : "linux"
```
Comment (SMillerDev) [inline Formula/atmos.rb:12]:
```suggestion
    ENV["GOOS"] = OS.kernel_name.downcase
```

File: Formula/atmos.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    # system "go", "get"
+    system "go build -o build/atmos -v -ldflags \"-X 'github.com/cloudposse/atmos/cmd.Version=test3'\""
+    bin.install "build/atmos"
```
Comment (SMillerDev) [inline Formula/atmos.rb:15]:
```suggestion
```

File: Formula/atmos.rb (hunk 1 above)
Comment (nitrocode) [reply Formula/atmos.rb]: Thanks. I force-pushed this change.

File: Formula/atmos.rb (hunk 3 above)
Comment (nitrocode) [reply Formula/atmos.rb]: Thanks. I force-pushed this change.

File: Formula/atmos.rb (hunk 2 above)
Comment (nitrocode) [reply Formula/atmos.rb]: Thanks. I force-pushed this change.

File: Formula/atmos.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  url "https://github.com/cloudposse/atmos/archive/refs/tags/v1.3.9.tar.gz"
+  sha256 "dabae8c89e508bd150640ec562f2642854c50a8e6ffbfd881a701ae40394f533"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOARCH"] = Hardware::CPU.intel? ? "amd64" : Hardware::CPU.arch.to_s
+    ENV["GOOS"] = OS.kernel_name.downcase
+    system "go", "build", *std_go_args(ldflags: "-X 'github.com/cloudposse/atmos/cmd.Version=#{version}'")
```
Comment (chenrui333) [inline Formula/atmos.rb:13]: we dont need to specify GOARCH or GOOS, as there is no cross-build happening with homebrew process
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w -X 'github.com/cloudposse/atmos/cmd.Version=#{version}'")
```

Comment (nitrocode) [reply Formula/atmos.rb:13]: @chenrui333 don't we have to adjust for linuxbrew users? or does it automatically set the correct environment variable when on linux or osx?

Comment (nitrocode) [reply Formula/atmos.rb:13]: there are also other formula that use GOOS and GOARCH env vars

https://github.com/Homebrew/homebrew-core/search?q=goos&type=

Comment (nitrocode) [reply Formula/atmos.rb]: @chenrui333 I made the above changes and force pushed.

**Review by SMillerDev — APPROVED** [review]
Thanks @nitrocode  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**